File: 0271-encode-and-decode-strings/0271-encode-and-decode-strings.cpp
class Codec {
public:

    // Encodes a list of strings to a single string.
    string encode(vector<string>& strs) {
        string encodedString = "";
        for (const string& s : strs){
            for (const char& c : s) {
                if (c == '/') encodedString += "//";
                else encodedString += c;
            }
            encodedString += "/:";
        }
        
        return encodedString;
    }

    // Decodes a single string to a list of strings.
    vector<string> decode(string s) {
        vector<string> decodedStrings;
        string currentString;
        
        for (size_t i = 0; i < s.size(); ++i){
            if (i < s.size() - 1 && s[i] == '/' && s[i + 1] == ':'){
                decodedStrings.push_back(currentString);
                currentString.clear();
                ++i;
            }
            else if (i < s.size() - 1 && s[i] == '/' && s[i + 1] == '/'){
                currentString += '/';
                ++i;
            }
            else currentString += s[i];
        }
        
        return decodedStrings;
    }
};

// Your Codec object will be instantiated and called as such:
// Codec codec;
// codec.decode(codec.encode(strs));

File: 0271-encode-and-decode-strings/0271-encode-and-decode-strings.py
class Codec:
    def encode(self, strs: List[str]) -> str:
        """Encodes a list of strings to a single string.
        """
        encoded_string = ""
        for s in strs:
            encoded_string += str(len(s)) + '/:' + s
        return encoded_string

    def decode(self, s: str) -> List[str]:
        """Decodes a single string to a list of strings.
        """
        decoded_strings = []
        i = 0
        
        while i < len(s):
            delim = s.find('/:', i)
            length = int(s[i:delim])
            str_ = s[delim + 2 : delim + 2 + length]
            decoded_strings.append(str_)
            i = delim + 2 + length
        return decoded_strings


# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(strs))